t/cmd/lfstest-badlocalfile.go

```diff
@@ -0,0 +1,72 @@
+//go:build testtools
+// +build testtools
+
+package main
+
+import (
+	"os"
+	"strings"
+
+	"github.com/git-lfs/pktline"
+)
+
+func main() {
+	pl := pktline.NewPktline(os.Stdin, os.Stdout)
+
+	// For any file Git asks us to clean, rename it and create a
+	// 0-byte file in its place. This allows Git to continue to
+	// read from the original file and send us its contents via
+	// packets on stdin, while our filter-process command will
+	// see the 0-byte file as soon as the first flush packet is
+	// received.
+	//
+	// If we simply truncate the file to 0 bytes in situ, Git
+	// may not be able to finish reading its full contents, and
+	// might send us fewer packets than actually needed.
```

Comment:
I don't know how to actually reproduce what Git is doing when it invokes our filter-process with incorrect files in the working copy, so I'm using this util to simulate it. Let me know if you figure out a way to reproduce this natively.